### DISCUSSION TITLE
why modules.d/multimap.conf not overwritten by override.d/multimap.conf

### DISCUSSION THREAD

**Discussion by OE2WNL** [docker-mailserver/docker-mailserver#3614] [Q&A]
I'm configuring black-/ and whitelist in my `override.d/multimap.conf` file which works as expected:

```shell
...
WHITELIST_IP {
    type = "ip";
    map = "/tmp/docker-mailserver/rspamd/whitelist_ip.map";
    description = "Local ip whitelist";
    action = "accept";
}
...
```

As I understand documentation when I define this file it would override the original `modules.d/multimap.conf` file.
But in the rspamd web ui I find Symbols for i.e.: FREEMAIL_ENVFROM which is only defined in the `modules.d/multimap.conf`

![grafik](https://github.com/docker-mailserver/docker-mailserver/assets/27812067/9caf2358-d376-4b3c-b207-75348c31cefd)

so the original file does not gets overwritten by my config but it gets merged ?
In this case it's what I want but just asking if at another part I don't want to merge config files.

Any input?

Thanks
-Wolfgang

**Comment by polarathene** [comment]
@georglauterbach may be able to assist when they have time.

Some of the rspamd integration requires using DMS specific configuration layer AFAIK, so this may require an update to our docs if not covered.

**Comment by georglauterbach** [comment]
> so the original file does not gets overwritten by my config but it gets merged ?

Indeed, Rspamd merges the files. Think about what would happen if the file would straight out overwrite the base configuration: forgetting a setting would lead to Rspamd becoming unusable.

Our documentation links to the Rspamd documentation in [this section](https://docker-mailserver.github.io/docker-mailserver/edge/config/security/rspamd/#manually). The part that says "[forcibly override](https://www.rspamd.com/doc/faq.html#what-are-the-locald-and-overrided-directories)" will lead you to the Rspamd documentation that clarifies what happens even further.

There, you can also read: "Types of settings which can be merged are collections (`{}`) and lists (`[]`); other settings would be effectively overridden by either file."

> In this case it's what I want but just asking if at another part I don't want to merge config files.

I cannot think of any place where you would not want this behavior :)

**Comment by OE2WNL** [comment]
From reading the doc's I assumed my `override.d/multimap.conf` would replace the `modules.d/multimap.conf` file completly. Re-reading the RSPAMD config makes it clearer.

**Usecase:**
In the modules.d/multimap.conf there are some parts configured i.e.
```
freemail_to {
    type = "header";
    header = "To";
    filter = "email:domain";
    map = "https://maps.rspamd.com/freemail/free.txt.zst";
    symbol = "FREEMAIL_TO";
    description = "To is a Freemail address";
    score = 0.0;
  }
```
What do I do if I don't want to have this `freemail_to` configuration at all?

**Reply by georglauterbach** [reply]
I understand what you want.

I guess the only benefit here is that you may save a lookup? The score already is 0, so this setting does not affect the overall spam score.
